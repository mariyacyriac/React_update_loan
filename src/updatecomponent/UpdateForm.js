

import React, { useState, useEffect } from 'react';
import useForm from './updatehooks/useForm';
import validate from './updatehooks/validateUpdateForm';
import 'bootstrap/dist/css/bootstrap.min.css';

const UpdateForm = (props) => {

    //const { handleChange, handleSubmit, errors ,resetAll, setDefaultVAlues} = useForm(validate); //, resetAll

   // console.log("props.location  from  test :::: " + JSON.stringify(props.location)); 

    const [loadLoanDetailsFrom, setLoadLoanDetailsFrom] = useState(
        props.location.loanDetailProps.loandetails
    );

    const { handleChange, handleSubmit, errors ,resetAll, setDefaultVAlues} = useForm(validate, loadLoanDetailsFrom);

  //  console.log("loadLoanDetailsFrom ::: " + JSON.stringify(loadLoanDetailsFrom));

    return (
        <div>
            <div className="col-12 col-md-6">
                <div className="card">
                    <div className="card-header">Loan Details Update Form</div>
                    <div className="card-body">
                        <form autoComplete="off" onSubmit={handleSubmit}>
                            <div className="form-group">
                                <label>Borrower name</label>
                                <input
                                    type="text"
                                    name="borrowerName"
                                    value={loadLoanDetailsFrom.borrowerName}
                                    className="form-control"
                                    disabled
                                />
                            </div>
                            <div className="form-group">
                                <label>Adderss line 1</label>
                                <input
                                    type="text"
                                    name="addressLine1"
                                    value={loadLoanDetailsFrom.addressLine1}
                                    className="form-control"
                                    disabled
                                />
                            </div>
                            <div className="form-group">
                                <label>Adderss line 2</label>
                                <input
                                    type="text"
                                    name="addressLine2"
                                    value={loadLoanDetailsFrom.addressLine2}
                                    className="form-control"
                                    disabled
                                />
                            </div>
                            <div className="form-group">
                                <label>City</label>
                                <input
                                    type="text"
                                    name="city"
                                    value={loadLoanDetailsFrom.city}
                                    className="form-control"
                                    disabled
                                />
                            </div>
                            <div className="form-group">
                                <label>Zipcode</label>
                                <input
                                    type="text"
                                    name="zipCode"
                                    value={loadLoanDetailsFrom.zipCode}
                                    className="form-control"
                                    disabled
                                />
                            </div>
                            <div className="form-group">
                                <label>Legal Documents</label>
                                <input
                                    type="text"
                                    name="legalDocuments"
                                    value={loadLoanDetailsFrom.legalDocuments}
                                    className="form-control"
                                    disabled
                                />
                            </div>
                            <div className="form-group">
                                <label>Loan Number</label>
                                <input
                                    type="text"
                                    name="loanNumber"
                                    value={loadLoanDetailsFrom.loanNumber}
                                    className="form-control"
                                    disabled
                                />
                            </div>




                            <div className="form-group">
                                <label>Loan Amount</label>
                                <input
                                    type="text"
                                    name="loanAmount"
                                    placeholder="Enter Loan amount"
                                    defaultValue={loadLoanDetailsFrom.loanAmount}
                                    className="form-control"
                                    onChange={handleChange}
                                />
                                <p className="text-danger">{errors.loanAmount}</p>

                            </div>
                            <div className="form-group">
                                <label>Loan Term</label>
                                <input
                                    type="text"
                                    name="loanTerm"
                                    placeholder="Enter Loan term "
                                    defaultValue={loadLoanDetailsFrom.loanTerm}
                                    className="form-control"
                                    onChange={handleChange}

                                />
                                <p className="text-danger">{errors.loanTerm}</p>


                            </div>
                            <div className="form-group">
                                <label>Lien Type</label>
                                <input
                                    type="text"
                                    name="lienType"
                                    placeholder="Enter Lien type"
                                    defaultValue={loadLoanDetailsFrom.lienType}
                                    className="form-control"
                                    onChange={handleChange}

                                />
                                <p className="text-danger">{errors.lienType}</p>

                            </div>

                            <div className="form-group">
                                <label>Lien ID</label>
                                <input
                                    type="text"
                                    name="lienId"
                                    placeholder="Enter lien id "
                                    defaultValue={loadLoanDetailsFrom.lienId}
                                    className="form-control"
                                   onChange={handleChange}

                                />
                                <p className="text-danger">{errors.lienId}</p>

                            </div>
                            <div className="form-group text-center">
                                <button className="btn btn-primary" style={{ margin: "5px" }}>Submit</button>
                                <button className="btn btn-danger" style={{ margin: "5px" }}  >Cancel</button>

                            </div>
                        </form>

                    </div>
                </div>

            </div>
        </div>
    );

};

export default UpdateForm;