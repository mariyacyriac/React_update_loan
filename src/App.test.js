// import React from 'react';
// import { render } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import UpdateForm from './updatecomponent/UpdateForm';


const loandetails = {
  loanAmount: 80000,
  loanTerm: 1.5,
  lienType: 'lien type A',
  lienId: 567,
  loanNumber: '123',
  borrowerName: 'abcd',
  addressLine1: 'street 122',
  addressLine2: 'house number 1',
  city: 'bangalore',
  zipCode: '90893',
  legalDocuments: 'doc1 , doc 2 , doc3',

}

const routeComponentPropsMock = {
  history: {},
  location: {
    loanDetailProps: { loandetails }
  },
  match: {},
}
const defaultProps = {
  ...routeComponentPropsMock
}
const wrapper = shallow(<UpdateForm {...defaultProps} />)

describe('test for rendering the update Compoent ', () => {
  it('should render the form', () => {
    expect(wrapper.getElement()).toMatchSnapshot()
  })
});

describe('test for rendering labels for the editable fields  ', () => {
  it("should have a label for loan Amount ", () => {
    const containerloanAmountLabel = shallow(<label>Loan Amount</label>);
    expect(containerloanAmountLabel).toMatchSnapshot();
  });

  it("should have a label for loan Term ", () => {
    const containerloanTermLabel = shallow(<label>Loan Term</label>);
    expect(containerloanTermLabel).toMatchSnapshot();
  });
  it("should have a label for Lien Type ", () => {
    const containerLienTypeLabel = shallow(<label>Lien Type</label>);
    expect(containerLienTypeLabel).toMatchSnapshot();
  });
  it("should have a label for Lien ID ", () => {
    const containerLienIDLabel = shallow(<label>Lien ID</label>);
    expect(containerLienIDLabel).toMatchSnapshot();
  });
});



describe('test for rendering buttons in update component ', () => {
  it("should have a submit button ", () => {
    const containerButtonSubmit = wrapper.find("button").at(0);
    expect(containerButtonSubmit.text()).toEqual("Submit");
  });
  it("should have a cancel button ", () => {
    const containerButtonSubmit = wrapper.find("button").at(1);
    expect(containerButtonSubmit.text()).toEqual("Cancel");
  });
});