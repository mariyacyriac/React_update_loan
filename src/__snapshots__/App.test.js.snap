// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for rendering labels for the editable fields   should have a label for Lien ID  1`] = `ShallowWrapper {}`;

exports[`test for rendering labels for the editable fields   should have a label for Lien Type  1`] = `ShallowWrapper {}`;

exports[`test for rendering labels for the editable fields   should have a label for loan Amount  1`] = `ShallowWrapper {}`;

exports[`test for rendering labels for the editable fields   should have a label for loan Term  1`] = `ShallowWrapper {}`;

exports[`test for rendering the update Compoent  should render the form 1`] = `
<div>
  <div
    className="col-12 col-md-6"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Loan Details Update Form
      </div>
      <div
        className="card-body"
      >
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label>
              Borrower name
            </label>
            <input
              className="form-control"
              disabled={true}
              name="borrowerName"
              type="text"
              value="abcd"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Adderss line 1
            </label>
            <input
              className="form-control"
              disabled={true}
              name="addressLine1"
              type="text"
              value="street 122"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Adderss line 2
            </label>
            <input
              className="form-control"
              disabled={true}
              name="addressLine2"
              type="text"
              value="house number 1"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              City
            </label>
            <input
              className="form-control"
              disabled={true}
              name="city"
              type="text"
              value="bangalore"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Zipcode
            </label>
            <input
              className="form-control"
              disabled={true}
              name="zipCode"
              type="text"
              value="90893"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Legal Documents
            </label>
            <input
              className="form-control"
              disabled={true}
              name="legalDocuments"
              type="text"
              value="doc1 , doc 2 , doc3"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Loan Number
            </label>
            <input
              className="form-control"
              disabled={true}
              name="loanNumber"
              type="text"
              value="123"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Loan Amount
            </label>
            <input
              className="form-control"
              defaultValue={80000}
              name="loanAmount"
              onChange={[Function]}
              placeholder="Enter Loan amount"
              type="text"
            />
            <p
              className="text-danger"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Loan Term
            </label>
            <input
              className="form-control"
              defaultValue={1.5}
              name="loanTerm"
              onChange={[Function]}
              placeholder="Enter Loan term "
              type="text"
            />
            <p
              className="text-danger"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Lien Type
            </label>
            <input
              className="form-control"
              defaultValue="lien type A"
              name="lienType"
              onChange={[Function]}
              placeholder="Enter Lien type"
              type="text"
            />
            <p
              className="text-danger"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Lien ID
            </label>
            <input
              className="form-control"
              defaultValue={567}
              name="lienId"
              onChange={[Function]}
              placeholder="Enter lien id "
              type="text"
            />
            <p
              className="text-danger"
            />
          </div>
          <div
            className="form-group text-center"
          >
            <button
              className="btn btn-primary"
              style={
                Object {
                  "margin": "5px",
                }
              }
            >
              Submit
            </button>
            <button
              className="btn btn-danger"
              onClick={[Function]}
              style={
                Object {
                  "margin": "5px",
                }
              }
            >
              Cancel
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
